contributors/design-proposals/storage/online-grow-volume-size.md

```diff
@@ -0,0 +1,344 @@
+<tr><td>Azure File</td><td>No</td><td>No</td><td>No</td><td>No</td></tr>
+<tr><td>Cephfs</td><td>No</td><td>No</td><td>No</td><td>No</td></tr>
+<tr><td>NFS</td><td>No</td><td>No</td><td>No</td><td>No</td></tr>
+<tr><td>Flex</td><td>Yes</td><td>Maybe</td><td>Maybe</td><td>No</td></tr>
+<tr><td>LocalStorage</td><td>Yes</td><td>Yes</td><td>No</td><td>No</td></tr>
+</table>
+
+## Implementation Design
+
+### Overview
+
+The key point of online file system resizing is how kubelet to discover which  PVCs need file system resizing. We achieve this goal by adding a `volumeFSResizingAnnotation` annotation to pod. In detail, if a volume received an online file system resizing request, the annotation will be added to the pods using this volume, so kubelet can watch for pods running on it to discover this annotation, and then start the online file system resizing operation for according volumes.
+
+The `volumeFSResizingAnnotation` annotation's key is `volumes.kubernetes.io/fs-resizing-{volumeName}`, and set it's value to `yes` if online file system resizing if required. We can use `volume.Spec.Name()` as the `volumeName` to guarantee that:
+
+* Annotation's key is a qualified name
+* `volumeName` is unique for a single pod
+
+### Prerequisite
+
+* `sig-api-machinery` need to allow pod's annotation update from kubelet. 
+* `sig-api-machinery` need to allow pod's get and annotation update from expand-controller.
+
+### Controller Manager Resize
+
+Expand controller is responsible to add the `volumeFSResizingAnnotation` to pods.
+
+#### Prerequisite of Controller Manager Resize
+
+* We use a three tuple `(pod,volume,node)` to represent a relationship: `pod` is using `volume`, `pod` is running on `node` and `volume` has been mounted to `node`. Add the `volumeFSResizingAnnotation` to `pod` only if this relationship was established.
+
+* For a specific `(volume,node)`, if more than one `pod` satisfies the relationship above, only add the annotation to one pod inside them.
+
+#### Fetch All Pods Using a PVC
+
+For a volume with online resizing request, we need to find which pods and nodes satisfy the relationship defines above, but `ExpandController` doesn't cache this relationship, so we need to fetch this information from `DesiredStateOfWorld` and `ActualStateOfWorld` of `AttachDetachController`. `DesiredStateOfWorld`caches the used volumes for each pods, `ActualStateOfWorld` caches the mounted volumes for each nodes.
+
+**Note**: `ActualStateOfWorld` only caches the mount relationship between volumes and nodes, not the volumes and pods, because only kubelet knows which pods actually mount which volumes.
+
+ To achieve this goal, we need to:
+
+* reflector the `pkg/controller/volume/attachdetach/cache` package to `pkg/controller/volume/cache`, and remove the parameter `volumePluginMgr` of `NewDesiredStateOfWorld` and `NewActualStateOfWorld` function, so that we can create a new `DesiredStateOfWorld` and `ActualStateOfWorld` without any dependence. 
```

Comment:
Do you mean `reflector` or you mean `refactor` ?

Reply:
Sorry, typo in word, I've fixed this.